src/components/ResourceList/ResourceList.tsx

```diff
@@ -84,82 +78,98 @@ export interface ResourceListProps {
   onSortChange?(selected: string, id: string): void;
   /** Callback when selection is changed */
   onSelectionChange?(selectedItems: ResourceListSelectedItems): void;
-  /** Function to render each list item	 */
-  renderItem(item: any, id: string, index: number): React.ReactNode;
+  /** Function to render each list item   */
+  renderItem(item: ItemType, id: string, index: number): React.ReactNode;
   /** Function to customize the unique ID for each item */
-  idForItem?(item: any, index: number): string;
+  idForItem?(item: ItemType, index: number): string;
   /** Function to resolve an id from a item */
-  resolveItemId?(item: any): string;
+  resolveItemId?(item: ItemType): string;
 }
 
-type CombinedProps = ResourceListProps & WithAppProviderProps;
-
-class ResourceListInner extends React.Component<CombinedProps, State> {
-  static Item = ResourceItem;
+type ResourceListType = (<ItemType>(
+  value: ResourceListProps<ItemType>,
+) => ReactElement) & {
```

Comment:
@BPScott Do you think `JSX.Element`?

Reply:
Looks like [JSX.Element is an alias to ReactElement](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/14dee133acca04c18ef4772661c8262313deb2ed/types/react/index.d.ts#L2946), so I think ReactElement is fine and good.

Stuff like [FuctionComponent](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/14dee133acca04c18ef4772661c8262313deb2ed/types/react/index.d.ts#L557) say their return type is ReactElement